CHANGELOG.md

```diff
@@ -5,6 +5,41 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+ * [`changed`] `base_config.inc` to work with I2C and SHDLC at the same time.
+               Changed HAL implemantation locations and moved the UART one from
+               the Makefile of the test in the sensor repo to here. Also changed
+               the way how mux sources are included so they are only needed by i2c
+               tests.
+ * [`added`] SHDLC implementation. This implementation was immigrated from the
+             `embedded-uart-sps` repo. Since it's general code which could be used
+             by multiple sensors it makes more sense to interate it here. The
+             Shdlc part also uses the files in `common/`. In addition to that
+             it adds two `.[ch]` file pairs. First `sensirion_shdlc.[ch]` which
+             contains the SHDLC implementation for Sensirion sensors. Sencondly
+             `sensirion_uart_hal.h` and `sensirion_uart_implementation.c` which
+             describe the hardware abstraction layer used by `sensirion_shdlc.c`.
+             This hal can be implemented in `sensirion_uart_implentation.c` for
+             the platform the sensor will be used on.
+ * [`changed`] Structure of embedded-common. File names should be more logical
+               and the spliting of the different parts of the code as well.
+               Embedded-common now splites the code into three `.[ch]` file pairs
+               and a configuration `.h` file.
+               First there are the three files in `common/`. `sensirion_config.h`
+               is similar to `sensirion_arch_config.h` from before. It contains
+               configurations for the code to be edited by the user.
+               Then there is the `sensirion_common.[ch]` pair which contains helper
+               functions which are not protocol spesific, at the moment it's mostly
+               type conversions.
+               Secondly there are the two other file pairs in `i2c/`. The pair
+               `sensirion_i2c.[ch]` contains all functions for the i2c
+               implementation of Sensirion. `sensirion_i2c_hal.h` describes the
+               hardware abstraction layer used by `sensirion_i2c.c`. This hal can
+               be implemented in `sensirion_hw_i2c_implementation.c` which is
+               located in `i2c/hw`. If no HW controller for I2C is present one can
+               also implement the GPIO SW-I2C implentation in
+               `i2c/sw/sensirion_sw_i2c_implementation.c`.
```

Comment:
Would it make sense to also put this in the `README.md`? That will also need an update anyway to reflect the changes

Reply:
Yeah true.